main.tf

```diff
@@ -220,6 +221,21 @@ resource "azurerm_kubernetes_cluster" "main" {
       condition     = !(var.microsoft_defender_enabled && !var.log_analytics_workspace_enabled)
       error_message = "Enabling Microsoft Defender requires that `log_analytics_workspace_enabled` be set to true."
     }
+    precondition {
+      // either we have automatic upgrades disabled
+      condition = (var.automatic_channel_upgrade == null || (
```

Comment:
The same [short-circuit](https://github.com/hashicorp/terraform/issues/24128) issue we've got in `variable`'s `validation` block.

Reply:
Should be solved now.